app/services/void_participant_declaration.rb

```diff
@@ -3,22 +3,30 @@
 class VoidParticipantDeclaration
   attr_accessor :cpd_lead_provider, :id
 
-  def call
-    declaration = ParticipantDeclaration.for_lead_provider(cpd_lead_provider).find(id)
+  def initialize(cpd_lead_provider:, id:)
+    @cpd_lead_provider = cpd_lead_provider
+    @id = id
+  end
 
+  def call
+    # TODO: these should be moved to activemodel validations
+    # and not controller level exceptions
```

Comment:
This does not seem to be a massive undertaking. Perhaps this could be done as part of the PR, as it is touching the validation.
However maybe the validations are fine here, this is not a controller, it is a service class and the validations here are tied to the process of voiding a declaration, there is an argument to be made for the validation to remain here I think.
if we'd move this validation to `active record` (I'm assuming here that's what you meant by `active model` we'd be coupling our database layer to the application logic, which is sometime more of pain than a helper.

Reply:
however I'm with you 💯 raising an error is a not ideal a simple error message added to the service class would be a bit cleaner

Reply:
nah, i'm doing this now. it would improve the code quality but makes zero difference in outcome of this ticket. I agree it should be done but I need to ship this change so we can do it later